src/provider/provider_level_zero.c

```diff
@@ -553,6 +481,132 @@ static umf_result_t ze_memory_provider_free(void *provider, void *ptr,
         g_ze_ops.zeMemFreeExt(ze_provider->context, &desc, ptr));
 }
 
+static umf_result_t query_min_page_size(ze_memory_provider_t *ze_provider,
+                                        size_t *min_page_size) {
+    assert(min_page_size);
+
+    LOG_DEBUG("Querying minimum page size");
+
+    void *ptr;
+    umf_result_t result = ze_memory_provider_alloc(ze_provider, 1, 0, &ptr);
+    if (result != UMF_RESULT_SUCCESS) {
+        return result;
+    }
+
+    ze_memory_allocation_properties_t properties = {
+        .stype = ZE_STRUCTURE_TYPE_MEMORY_ALLOCATION_PROPERTIES};
+    ze_result_t ze_result = g_ze_ops.zeMemGetAllocProperties(
+        ze_provider->context, ptr, &properties, NULL);
+
+    *min_page_size = properties.pageSize;
+
+    ze_memory_provider_free(ze_provider, ptr, 1);
+
+    return ze2umf_result(ze_result);
+}
+
+static void ze_memory_provider_finalize(void *provider) {
+    ze_memory_provider_t *ze_provider = (ze_memory_provider_t *)provider;
+    umf_ba_global_free(ze_provider->resident_device_handles);
+
+    umf_ba_global_free(provider);
+}
+
+static umf_result_t ze_memory_provider_initialize(void *params,
+                                                  void **provider) {
+    if (params == NULL) {
+        return UMF_RESULT_ERROR_INVALID_ARGUMENT;
+    }
+
+    umf_level_zero_memory_provider_params_handle_t ze_params =
+        (umf_level_zero_memory_provider_params_handle_t)params;
+
+    if (!ze_params->level_zero_context_handle) {
+        LOG_ERR("Level Zero context handle is NULL");
+        return UMF_RESULT_ERROR_INVALID_ARGUMENT;
+    }
+
+    if ((ze_params->memory_type == UMF_MEMORY_TYPE_HOST) ==
+        (ze_params->level_zero_device_handle != NULL)) {
+        LOG_ERR("Level Zero device handle is NULL");
```

Comment:
The log message is incorrect if both conditions are `true`.

Reply:
Done.